liblumen_alloc/src/erts/term/map.rs

```diff
@@ -81,10 +81,12 @@ impl Map {
     }
 
     pub fn update(&self, key: Term, value: Term) -> Option<HashMap<Term, Term>> {
-        let mut map = self.value.clone();
-        match map.insert(key, value) {
-            Some(_) => Some(map),
-            None => None,
+        if self.get(key).map_or(false, |v| v == value) {
```

Comment:
I thought I had put this into the update/3 PR, but noticed it was sitting uncommitted in my working directory.